docker/local-development.md

```diff
@@ -0,0 +1,279 @@
+# Docker for Local Development
+
+Docker for local development was piloted on `dedupe-service`, a repository to
+which some DataMade employees do not have access. Luckily, we can abstract general
+patterns for Dockerizing an application to guide our next effort.
+
+## Overview
+
+A containerized local development environment has four components, and one
+optional service:
+
+1. [A Dockerfile](#1-dockerfile), containing building instructions for the application itself.
+2. [A database initialization script](#2-scriptsinit-dbsh)
+that creates your database and installs any extensions
+3. [A root `docker-compose.yml` file](#3-docker-composeyml)
+that declares the application and its dependent services
+4. [A `tests/docker-compose.yml` file](#4-testsdocker-composeyml)
+that overrides the application service in the root file, in order to run the tests
+4. [A `docker-compose.db-ops.yml` file](#5-docker-composedb-opsyml-optional)
+that automates a multi-step data loading routine (Optional)
+
+With this setup, you can:
+
+- [Run your application or tests](#run-the-application)
+- [Run custom commands on your container](#run-custom-commands-on-containers)
+
+## Components
+
+### 1. `Dockerfile`
+
+```Dockerfile
+# Extend the base Python image
+# See https://hub.docker.com/_/python for version options
+# N.b., there are many options for Python images. We used the plain
+# version number in the pilot. YMMV. See this post for a discussion of
+# some options and their pros and cons:
+# https://pythonspeed.com/articles/base-image-python-docker-images/
+FROM python:<PYTHON_VERSION>
```

Comment:
Developers will probably want to copy/paste this file, so maybe we should include a default here?

```suggestion
FROM python:3.7
```